src/AbstractRecursiveIterator.php

```diff
@@ -38,8 +38,9 @@
      */
     protected function _construct()
     {
-        $this->_resetParents();
         parent::_construct();
+
+        $this->_resetParents();
```

Comment:
This is not a cleanup. You have switched the order of these operations. Why?

Reply:
Isn't that how it should be? The parent's parameterless constructor should be called first when chaining, no?

Reply:
Not really. If there's a need for that, then yes. But I think that first, the class should do what it knows about, then let the ancestors do their thing. At least, that's how I do it everywhere.

Reply:
I see. Intuitively I'd think the opposite. Let ancestor do its thing, then the class gets its turn. I reverse them again?

Reply:
Reverse pls, so that at least it's consistent everywhere.